src/routes/Home.tsx

```diff
@@ -0,0 +1,99 @@
+import { Button, Input, Text, ChakraProvider, Box, Spacer, HStack, VStack, useToast } from "@chakra-ui/react";
+import { extendTheme } from "@chakra-ui/react";
+import React from 'react';
+import '/src/App.css';
+
+const theme = extendTheme({
+    textStyles: {
+      h1: {
+        fontSize: ['48', '72px'],
+        lineHeight: '110%',
+        letterSpacing: '-2%',
+        color:"white",
+        fontFamily:"Kufam",
+        top:"20%",
+      },
+      h2: {
+        fontSize: ['36px', '48px'],
+        fontWeight: 'semibold',
+        lineHeight: '110%',
+        letterSpacing: '-1%',
+        fontFamily:"Kufam",
+        padding:10,
+
+      },
+    },
+  })
+
+export default function Home () {
+    const toast = useToast()
+    const [email, setEmail] = React.useState('')
+    const handleChange = (event: { target: { value: React.SetStateAction<string>; }; }) => setEmail(event.target.value)
+
+    const handleClickToast = async () => {
+        try {
+            console.log(email)
+          const response = await fetch('http://18.119.140.95:3000/subscription', {
+            method: 'POST',
+            headers: {
+              'Content-Type': 'application/json',
+            },
+            body: JSON.stringify({
+                'email':email,
+                'mailingList':'rp_interest'
+            }),
+          });
+    
+          if (!response.ok) {
+            throw new Error('Failed to fetch data');
+          }
+        //   const data = await response.json();
+          toast({
+            title: 'Sucess!',
+            description: 'You have been added to the R|P mailing list',
+            status: 'success',
+            duration: 3000,
+            isClosable: true,
+          });
+        } catch (error) {
+        //   console.error('Error fetching data:', error);
+          toast({
+            title: 'Sorry try again',
+            description: 'An error occurred while adding you.',
+            status: 'error',
+            duration: 3000,
+            isClosable: true,
+          });
+        }
+      };
```

Comment:
you don't need to do all this! can simplify it a lot :)
```suggestion
          const promise = fetch('http://18.119.140.95:3000/subscription', {
            method: 'POST',
            headers: {
              'Content-Type': 'application/json',
            },
            body: JSON.stringify({
                'email':email,
                'mailingList':'rp_interest'
            }),
          });
          
          toast.promise(promise, {
          success: { title: 'Success!', description: 'You\'ve been added :)' },
          error: { title: 'Oops!', description: 'Something went wrong - please try again' },
          loading: { title: '<INSERT TITLE>', description: 'Please wait' },
        })
    ```

Reply:
I believe you have to do it that way because the api request isn't a Promise type, which is what the toast.promise takes in as a parameter, and from what I know you can't convert one to another a different way.